snorql-extensions/src/main/kotlin/com/udaan/snorql/extensions/storage/metrics/DbIndexRedundancyMetric.kt

```diff
@@ -0,0 +1,273 @@
+        val childIncludeCols = childIndex.includeColumnNrs?.split(" ")?.toSet()
+        logger.debug(
+            "[DbIndexRedundancyMetric][isDuplicate] Parent Include Cols: $parentIncludeCols\n" +
+                    "Child Include Cols: $childIncludeCols"
+        )
+        if (parentIndex.indexColumnNrs == childIndex.indexColumnNrs && parentIncludeCols == childIncludeCols) {
+            return true
+        }
+        return false
+    }
+
+    private fun isOverlapping(parentIndex: DbIndexRedundancyDTO, childIndex: DbIndexRedundancyDTO): Boolean {
+        val parentIncludeCols =
+            parentIndex.includeColumnNrs?.split(" ")?.filterNot { it.isNullOrBlank() }?.toSet() ?: setOf()
+        val childIncludeCols =
+            childIndex.includeColumnNrs?.split(" ")?.filterNot { it.isNullOrBlank() }?.toSet() ?: setOf()
+        logger.debug(
+            "[DbIndexRedundancyMetric][isDuplicate] Parent Include Cols: $parentIncludeCols\n" +
+                    "Child Include Cols: $childIncludeCols"
+        )
+        if (checkStartsWithWord(
+                parentIndex.indexColumnNrs,
+                childIndex.indexColumnNrs
+            ) // && (parentIncludeCols subtract childIncludeCols).isEmpty()
+            && (childIncludeCols subtract parentIncludeCols).isEmpty()
+        ) {
+            return true
+        }
+        return false
+    }
+
+    private fun isSimilar(parentIndex: DbIndexRedundancyDTO, childIndex: DbIndexRedundancyDTO): Boolean {
+        if (parentIndex.indexColumnNrs == childIndex.indexColumnNrs
+            && parentIndex.includeColumnNrs != childIndex.includeColumnNrs
+        ) {
+            return true
+        }
+        return false
+    }
+
+    override fun getMetricResponseMetadata(
+        metricInput: DbIndexRedundancyInput,
+        metricOutput: MetricOutput<DbIndexRedundancyResult, IMetricRecommendation>
+    ): Map<String, Any>? {
+        val responseMetadata = mutableMapOf<String, Any>()
+        val metricConfig = getMetricConfig(metricInput.metricId)
+        val query = metricConfig.queries["main"]
+        responseMetadata["underlyingQueries"] = listOf(query)
+        responseMetadata["referenceDocumentation"] = metricConfig.referenceDoc
+        responseMetadata["description"] = metricConfig.description
+        responseMetadata["supportsHistorical"] = metricConfig.supportsHistorical
+        return responseMetadata
+    }
+
+    private fun checkStartsWithWord(longerString: String?, shorterString: String?): Boolean {
+        if (longerString == null) return false
+        if (shorterString == null) return true
+        val longerArr = longerString.split(" ")
+        val shortArr = shorterString.split(" ")
+        if (longerArr.size < shortArr.size) return false
+        shortArr.forEachIndexed { index, element ->
+            if (element != longerArr[index]) {
+                return false
+            }
+        }
+        return true
```

Comment:
This seems too complicated. Please use kotlin methods.

Reply:
Can you please suggest optimisations on this?